smithy-model/src/main/java/software/amazon/smithy/model/transform/DowngradeToV1.java

```diff
@@ -119,12 +118,8 @@ private Model removeUnnecessaryDefaults(ModelTransformer transformer, Model mode
     private Model removeOtherV2Traits(ModelTransformer transformer, Model model) {
         Set<Shape> updates = new HashSet<>();
 
-        for (StructureShape structure : model.getStructureShapes()) {
-            for (MemberShape member : structure.getAllMembers().values()) {
-                if (member.hasTrait(ClientOptionalTrait.class)) {
-                    updates.add(member.toBuilder().removeTrait(ClientOptionalTrait.ID).build());
-                }
-            }
+        for (MemberShape member : model.getMemberShapesWithTrait(ClientOptionalTrait.class)) {
```

Comment:
This trait can only be applied to structure member, so why check all members?

Reply:
Reverted the change.